### ISSUE TITLE
Getting three warning messages after upgrading to 4.10.0-0.okd-2022-04-23-131357

### ISSUE THREAD

**Issue by johnlongo** [openshift/okd#1195]
Getting three warning messages after upgrading to 4.10.0-0.okd-2022-04-23-131357. Please advise what to do to resolve.

![image](https://user-images.githubusercontent.com/23619505/164955213-af05852e-cc34-4445-9d24-91eb97e91428.png)

Apr 23, 2022, 9:46 PM
Deprecated API that will be removed in the next EUS version is being used. Removing the workload that is using the discovery.k8s.io.v1beta1/endpointslices API might be necessary for a successful upgrade to the next EUS cluster version. Refer to `oc get apirequestcounts endpointslices.v1beta1.discovery.k8s.io -o yaml` to identify the workload.

oc get apirequestcounts endpointslices.v1beta1.discovery.k8s.io -o yaml
apiVersion: apiserver.openshift.io/v1
kind: APIRequestCount
metadata:
  creationTimestamp: "2021-12-25T12:57:03Z"
  generation: 1
  name: endpointslices.v1beta1.discovery.k8s.io
  resourceVersion: "65480484"
  uid: bfa9417d-be2e-4009-bfd7-6a2a42d91a9d
spec:
  numberOfUsersToReport: 10
status:
  currentHour:
    byNode:
    - byUser:
      - byVerb:
        - requestCount: 2
          verb: watch
        requestCount: 2
        userAgent: master0.ocp4-2.apptecinc.com/ovnkube@c580607cf5f3
        username: system:serviceaccount:openshift-ovn-kubernetes:ovn-kubernetes-controller
      nodeName: 192.168.5.40
      requestCount: 2
    - nodeName: 192.168.5.41
      requestCount: 0
    - nodeName: 192.168.5.42
      requestCount: 0
    requestCount: 2

Apr 23, 2022, 9:41 PM
Deprecated API that will be removed in the next EUS version is being used. Removing the workload that is using the batch.v1beta1/cronjobs API might be necessary for a successful upgrade to the next EUS cluster version. Refer to `oc get apirequestcounts cronjobs.v1beta1.batch -o yaml` to identify the workload.

oc get apirequestcounts cronjobs.v1beta1.batch -o yaml
apiVersion: apiserver.openshift.io/v1
kind: APIRequestCount
metadata:
  creationTimestamp: "2021-12-25T13:12:01Z"
  generation: 1
  name: cronjobs.v1beta1.batch
  resourceVersion: "65489205"
  uid: b9cf8c6c-bec4-4b5f-8ead-087c80627e36
spec:
  numberOfUsersToReport: 10
status:
  currentHour:
    byNode:
    - byUser:
      - byVerb:
        - requestCount: 4
          verb: watch
        requestCount: 4
        userAgent: percona-server-mongodb-operator/v0.0.0
        username: system:serviceaccount:mongodb:percona-server-mongodb-operator
      nodeName: 192.168.5.40
      requestCount: 4
    - byUser:
      - byVerb:
        - requestCount: 4
          verb: watch
        requestCount: 4
        userAgent: v2.3.0
        username: system:serviceaccount:openshift-monitoring:kube-state-metrics
      - byVerb:
        - requestCount: 4
          verb: watch
        requestCount: 4
        userAgent: postgres-operator/v0.0.0
        username: system:serviceaccount:pgo:pgo
      nodeName: 192.168.5.41
      requestCount: 8
    - nodeName: 192.168.5.42
      requestCount: 0
    requestCount: 12

Apr 23, 2022, 9:41 PM
Deprecated API that will be removed in the next EUS version is being used. Removing the workload that is using the policy.v1beta1/poddisruptionbudgets API might be necessary for a successful upgrade to the next EUS cluster version. Refer to `oc get apirequestcounts poddisruptionbudgets.v1beta1.policy -o yaml` to identify the workload.

oc get apirequestcounts poddisruptionbudgets.v1beta1.policy -o yaml
apiVersion: apiserver.openshift.io/v1
kind: APIRequestCount
metadata:
  creationTimestamp: "2021-12-25T12:55:14Z"
  generation: 1
  name: poddisruptionbudgets.v1beta1.policy
  resourceVersion: "65497166"
  uid: 6f7135d9-d4b4-45ae-b295-3f915efc3ab3
spec:
  numberOfUsersToReport: 10
status:
  currentHour:
    byNode:
    - byUser:
      - byVerb:
        - requestCount: 11
          verb: get
        - requestCount: 11
          verb: update
        requestCount: 22
        userAgent: cluster-network-operator/v0.0.0
        username: system:serviceaccount:openshift-network-operator:default
      - byVerb:
        - requestCount: 4
          verb: watch
        requestCount: 4
        userAgent: percona-server-mongodb-operator/v0.0.0
        username: system:serviceaccount:mongodb:percona-server-mongodb-operator
      nodeName: 192.168.5.40
      requestCount: 26
    - byUser:
      - byVerb:
        - requestCount: 4
          verb: watch
        requestCount: 4
        userAgent: v2.3.0
        username: system:serviceaccount:openshift-monitoring:kube-state-metrics
      nodeName: 192.168.5.41
      requestCount: 4
    - nodeName: 192.168.5.42
      requestCount: 0
    requestCount: 30

**Comment by jacksgt** [comment]
Hey there,
I can confirm that we're also getting three alerts on a fresh OKD 4.10 cluster.

However, these API requests are generated by kube-state-metrics/ArgoCD/kube-controller-manager, since these components poll all available APIs.
The API is not actually used by anything, thus the alert can be ignored.

```yaml
#oc get apirequestcounts poddisruptionbudgets.v1beta1.policy -o yaml
apiVersion: apiserver.openshift.io/v1
kind: APIRequestCount
metadata:
  creationTimestamp: "2022-03-28T07:45:21Z"
  generation: 1
  name: poddisruptionbudgets.v1beta1.policy
  resourceVersion: "27359338"
  uid: 156c04bf-6133-4fb7-a243-8d92e4abac4f
spec:
  numberOfUsersToReport: 10
status:
  currentHour:
    byNode:
    - byUser:
      - byVerb:
        - requestCount: 5
          verb: watch
        requestCount: 5
        userAgent: v2.3.0
        username: system:serviceaccount:openshift-monitoring:kube-state-metrics
      nodeName: 0.0.0.0
      requestCount: 5
    requestCount: 5
```

```yaml
# oc get apirequestcounts cronjobs.v1beta1.batch -o yaml
apiVersion: apiserver.openshift.io/v1
kind: APIRequestCount
metadata:
  creationTimestamp: "2022-03-28T07:47:47Z"
  generation: 1
  name: cronjobs.v1beta1.batch
  resourceVersion: "27354458"
  uid: e093f9b2-a439-4f1a-8986-b20a959ce805
spec:
  numberOfUsersToReport: 10
status:
  currentHour:
    byNode:
    - byUser:
      - byVerb:
        - requestCount: 3
          verb: watch
        requestCount: 3
        userAgent: v2.3.0
        username: system:serviceaccount:openshift-monitoring:kube-state-metrics
      nodeName: 0.0.0.0
      requestCount: 3
    requestCount: 3
```

```yaml
# oc get apirequestcounts podsecuritypolicies.v1beta1.policy -o yaml
apiVersion: apiserver.openshift.io/v1
kind: APIRequestCount
metadata:
  creationTimestamp: "2022-03-28T07:18:02Z"
  generation: 1
  name: podsecuritypolicies.v1beta1.policy
  resourceVersion: "27361068"
  uid: 57f67945-c515-4445-b577-a290c08dfde4
spec:
  numberOfUsersToReport: 10
status:
  currentHour:
    byNode:
    - byUser:
      - byVerb:
        - requestCount: 6
          verb: list
        - requestCount: 6
          verb: watch
        requestCount: 12
        userAgent: argocd-application-controller/v0.0.0
        username: system:serviceaccount:openshift-cern-argocd:argocd-application-controller
      - byVerb:
        - requestCount: 5
          verb: watch
        requestCount: 5
        userAgent: kube-controller-manager/v1.23.5
        username: system:kube-controller-manager
      nodeName: 0.0.0.0
      requestCount: 17
    requestCount: 17
```

You can check if you actually have any of these deprecated API in your cluster by running:
```sh
oc get -A podsecuritypolicies.v1beta1.policy
Warning: policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+
No resources found
```

It should say "No resources found".

I also wrote a blog post about identifying usage of deprecated APIs with ArgoCD / Openshift:
https://blog.cubieserver.de/2022/using-argocd-to-find-kubernetes-resources-using-deprecated-api-versions/

**Comment by vrutkovs** [comment]
>username: system:serviceaccount:openshift-network-operator:default
>username: system:serviceaccount:openshift-monitoring:kube-state-metrics

These are openshift bugs and would need to be resolved before we upgrade to 4.12

>username: system:serviceaccount:mongodb:percona-server-mongodb-operator
This operator needs to be updated before 4.12 upgrade
